bin/src/config.rs

```diff
@@ -77,9 +81,39 @@ impl Check {
             Ok(ReadOnlyVfs::singleton("<stdin>", src.as_bytes()))
         } else {
             let all_ignores = [self.ignore.as_slice(), extra_ignores].concat();
-            let ignore = dirs::build_ignore_set(&all_ignores, &self.target, self.unrestricted)?;
-            let files = dirs::walk_nix_files(ignore, &self.target)?;
-            vfs(files.collect::<Vec<_>>())
+
+            // TODO: A more modern `clap` may be able to handle this
+            // without falling back to a destructure-then-restructure
+            // with the help of a `ValueEnum`.
+            //
+            // There are definitely less clear ways of expressing this
+            // that avoid the deep copy, but this should not be
+            // particularly expensive compared to the actual work.
+            let files = match &self.targets[..] {
+                [target] if target.is_dir() => {
+                    // If we only have one target, and it is a
+                    // directory, "do what I mean", that is, just run
+                    // statix on everything but respect the
+                    // `.gitignore` file and `--ignore` flags.
+                    //
+                    // We specifically don't follow this behavior if
+                    // the user specifies exactly one file, because
+                    // ignoring the only file we specify doesn't
+                    // really make sense, and it makes integrating
+                    // with third party scripting easier.
+                    let gitignore_path = target.join(".gitignore");
+                    let ignore =
+                        dirs::build_ignore_set(&all_ignores, &gitignore_path, self.unrestricted)?;
+                    dirs::walk_nix_files(ignore, target)?.collect()
+                }
+                targets @ [..] => {
+                    // Otherwise, assume the user specified exactly those
+                    // paths they want to check
+                    targets.to_vec()
```

Comment:
I think we're quite clear about this in the docs, but a warning if the ignore flags are used may be appropriate?